### DISCUSSION TITLE
RDT Restricted Data Token for PII

### DISCUSSION THREAD

**Discussion by smithburger** [abuzuhri/Amazon-SP-API-CSharp#245] [General]
Does your library support the new RDT Restricted Data Token which is effective/required on July 31st, 2022 to access PII? I was looking over the source code and didn't see anything about it but maybe I am missing something? I didn't see any previous discussion on this.

From Amazon.

**Why are you receiving this email?**
Effective July 31, 2022, Vendors who are currently live with Direct Fulfilment (DF) Selling Partner APIs are required to use a new access protocol via the Tokens API to make successful API calls to restricted operations. 

**What is changing?**
Operations that return restricted data (such as Personally Identifiable Information) are considered restricted operations, and require special authorization in the form of a Restricted Data Token (RDT). For these operations, RDT will be replacing the LWA access token in the API requests.

**DF API operations that will require the use of RDT are the following:
Orders API:**
•	getOrders (using date range)
•	getOrder (using a PO#)
**Shipment API:**
•	getShippingLabels (using date range)
•	getShippingLabel (using a PO#)
•	getPackingSlips (using date range)
•	getPackingSlip (using a PO#)
•	getCustomerInvoices (using a date range)
•	getCustomerInvoice (using a PO#)

**Why is this changing?**
Amazon takes protecting Customer data seriously. This change is being made to ensure vendors are using the latest technology and features that provides enhanced security when trying to access restricted information from Amazon’s APIs.

**What action is required by the vendor?**
If you are using an application that is affected by this change, you must update your application to use the new access protocol by July 31, 2022, to avoid service disruption as LWA access token will no longer work for restricted operations after this date.

**How can vendors generate and Test RDT:**
For more information on how to use RDT, please refer to the attached “RDT Generation and Testing Guide” and the Github[ link](https://developer-docs.amazon.com/sp-api/docs/tokens-api-use-case-guide). Also, attached is a zip file with supporting documents (postman collection, sample code files in C# and Python) to help you simulate and “test” the API operations using RDT. Please note that you should test directly pointing to our production endpoints.

**Comment by smithburger** [comment]
I think I missed it and you have this built in already with CreateRestrictedDataTokenRequest? If so you can close this discussion and my apologies. I tried to search the repository and nothing came up. Missed that example.

**Reply by abuzuhri** [reply]
Yes that true we are support RDT
